Source/WebCore/page/Quirks.cpp

```diff
@@ -359,6 +360,30 @@ bool Quirks::shouldAvoidUsingIOS17UserAgentForFacebook() const
 #endif
 }
 
+HashMap<RegistrableDomain, UserAgentStringQuirk>& Quirks::updatableStorageAccessUserAgentStringQuirks()
+{
+    static NeverDestroyed<HashMap<RegistrableDomain, UserAgentStringQuirk>> map;
+    return map.get();
+}
+
+void Quirks::setStorageAccessUserAgentStringQuirks(Vector<UserAgentStringQuirk>&& userAgentStringQuirks)
+{
+    auto& quirks = updatableStorageAccessUserAgentStringQuirks();
+    quirks.clear();
+    for (auto&& quirk : userAgentStringQuirks)
+        quirks.set(quirk.domain, quirk);
```

Comment:
This should be a call to `add()`, not `set()` I think. Since you've just cleared the HashMap, there shouldn't be any need to overwrite. There is a possibility that there are several quirks for the same domain in the vector I guess. Not sure if that's possible but even if it is, it is unclear using the last matching quirk instead of the first matching one is better.

add() is more efficient.